### DISCUSSION TITLE
CAT Control and Noise cancelling

### DISCUSSION THREAD

**Discussion by f4bpp** [AlexandreRouma/SDRPlusPlus#1117] [General]
Hello,
My name is David and I'm a French Ham Radio Operator (F4BPP). I own a Yaesu FT-847 transceiver with the TAP 50 SDR kit which provide an FI output for connecting it to a sdr receiver. My computer is running Kubuntu Jammy Linux OS. I've tried many SDR software like GQRX (to limited for me), Cubic SDR (poor user interface and lack of functionality), SDRAngel (a real gas mill very powerful... too powerful for me with non intuitive and easy interface and very hugh latency). For me, the best of the best is SDR++ and first of all I want to congratulate Alexandre for his amazing work. 

The only lack of SDR++ is that I can't control my transceiver with it. Many people use SDR software to add waterfall to their ol radio like me and the very needed feature for all of us is to change the radio frequency when we select one in the SDR software with the mouse. In this way, when we find a station on the scope we just have to click on it then take the radio microphon and talk with this station. Actually I have to enter the frequency by hand and this is very annoying especially when you have to do it several times in a short space of time.

So I want this feature to be added and if I can help, I'd be delighted to do so. I've made a python software to init the Hamlib connexion between my transceiver and the computer so the software like Gpredict can send the frequency to the radio by using the standard and common Rigctl commands to do that.

And finally, the icing on the cake would be to have a noise cancelling function like the one on SDR Console and this software would be perfect for me. No need to reinvent the wheel and add a ton of modules to do do what other software already do like APT and DMR decoding, Satellite tracking etc...  

So I hope my request will be taken into account because I don't want to switch to Windows just for using a SDR software that do the job.

Thanks for reading and best regards, David.

**Comment by AlexandreRouma** [comment]
You already can but it's still in early development.
First make sure you're running the latest SDR++ nightly build (https://www.sdrpp.org/nightly)
Start hamlib rigctld and use the RigCTL Client module -> In the "Module Manager" menu -> select "rigctl_client" from the dropdown at the bottom of the menu and type any name you want for it in the text box next to said dropdown (eg. "RigCTL Client") then click que "+" button to add it. A new menu entry will appear at the bottom (you can drag it up where you want it to be). Then you can connect to hamlib using it.

Do note that again, it's in early development. The connection only currently works from SDR++ to the rig, not yet the other way around (it won't follow if you move the VFO on your rig).

**Comment by f4bpp** [comment]
Many thanks for your quick answer Alexandre. I've already tried this before I open this new discussion. It does not work and I would be happy to help you fixing the issue.

I start the connection between my transceiver and the computer with this command line in my python program :

rigctld -m 1001 -r /dev/ttyUSB.YAESU-DIGIMODE -t 4532 -s 57600 -P RTS --set-conf=stop_bits=2,serial_handshake=None

SDR++ display a green message "Connected" and the "CAT" indicator is played on myhe transceiver so I know the connection is operational. Another thing is that when I connect the client, the frequency number don't change but I lost audio as if I were changing the frequency  and the waterfall seems to shift to the left. I think I think it's because of  the IF frequency value but I don't know which one I have to put in this field. Actually I'm using a RTL-SDR dongle V3 but I will soon receive the Airspy HF+ discovery.

I hope you will help me to make it work.

Thanks in advance, David.

**Reply by AlexandreRouma** [reply]
Could you explain exactly in what way it doesn't work? I'm not sure I understand.
The IF value is the fixed frequency at which the SDR will stay tuned. according to what I can find for the TAP50, this should be 50MHz (`50000000`). When the center frequency displayed in SDR++ changed, the rigctl client module will tell your yaesu radio to tune instead of tuning the SDR.

**Reply by f4bpp** [reply]
When you use the mode symbolised by the target icon, it works, but when you use the mode symbolised by the two arrows icon, it no longer works. Yet this is the only mode I use.

 I don't understand why we must enter the IF frequency for this function. I didn't find the one who match my SDR receiver connected to the IF output of my Yaesu FT-847 sdr kit interface. So, when I connect Rigctl, because I don't know this frequency, the waterfall shift and I can't ear anymore the frequency I was listening before I click on the start button. 

You should remove the IF frequency field for this function or, if it is useful for someone, add the possibility to enable it if needed with a checkbox. 

So please help a technology teacher working in a French middle school who makes educational project about Hamradio on Linux computers. I really needs SDR++ to control my radio. We have 2 computer rooms at school with 32 machines in each one running Linux. I have recycled the 64 old computers with Linux tuned like Windows 11 to save money. And the best SDR software I've found for this OS is SDR++.

**Comment by f4bpp** [comment]
Hi Alexandre.

In case my last mail would be considered as a spam again, I write here my answer to your last message.
I'm using the PAT V SDR interface to connect ma radio to the SDR receiver and it act in a specific way. Let me explain you how my configuration works : 

First, I have to select on my transceiver, the band I want to work with. So, in order to allow SDR++ to receive all the frequencies of a chosen band, I have to select this band on my transceiver.

I have a button for that to switch between 3, 7, 10, 14, 18, 21, 24, 28, 50, 140 and 430 MHz. I cant either enter a frequency with a numpad so the transceiver switch to the band to which the frequency belongs. So, when I select a frequency on SDR++, I have to select the band to which the frequency belongs on my transceiver because if I don't, SDR++ doesn't receive anything.

But, when I select a Hamradio band on my radio, whatever the frequency I choose on it, SDR++ is able receive the whole frequencies of all that band and beyond. There is no need to set the same frequency on both my transceiver and SDR++.

That means that if SDR++ send a frequency to the transceiver via the CAT port, the transceiver will switch the according band and SDR++ will receive all the frequencies of this band.So keep in mind that there is no need to deal with the IF frequency to do that.

I use SDR++ for the waterfall of all the bandwith according to the band I select on the transceiver. So I can select the band with with the numpad of my radio but if I could, I would just send the frequency to the radio via the Rigctl client module of SDR++ to get the same result.

All that I need is that when I select a frequency by clicking on the waterfall, whatever the mode (double arrow or target), SDR++ send this frequency to the radio through the CAT port via Rigctl so the transceiver switch to the right band.

I can't transmit with SDR++ since it act as a receiver. When I see one transmission on the waterfall I click on it and the SDR++ frequency is tune to where I have clicked, that’s the deal. But after I have to enter this frequency to my radio too. I just want to avoid having to enter this frequency to my transceiver and for htat, I need that SDR++ send to the transceiver the frequency I have clicked on to the transceiver without changing the IF frequency, nothing else. 

Actually, SDR++ do that if we use the target mode after Rigctl is engaged but, in the same time, it changes the frequency but there is no need to do that. The only result is that I loose the signal because the frequency have changed. The target mode is not useful for me because I always select a frequency without moving the band I have extended with the zoom, so that it took up the whole width of the screen.

To sum up, I would like SDR++ act like SDR Console on windows : When I select a frequency on SDR++, it send this one to the transceiver. This allow me to select easily a frequency with the SDR software, and when I want to talk with the person transmitting on this frequency I have selected on SDR++ (whatever the way), I just have to take the microphone of my radio, press the PTT and talk. SDR++ as receiver and my radio as transmitter. 

I assure you that there is no need to change anything, not even the IF frequency to do that. You just have to send the frequency to the radio with CAT port via Rigct, nothing else.

Please, could you try do do that for me ?

Many thanks for time passed to read all my messages. I know I ask you a lot but you are my last chance to make my educational project with my schoolchildren in the middle school I’m working in next September. I will explain you later why it is so important for me and for the students that you try to code the Rigctl client to fit my needs. It's not only for my pleasure.

Best regards, David.
